20230311/mike.java

```diff
@@ -0,0 +1,34 @@
+/*
+Use slow and fast pointer to find mid node(just like recent questions).
+If we find mid node, cut it into two part as left subtree and right subtree.
+*/
+class Solution {
+    public TreeNode sortedListToBST(ListNode head) {
+        if(head==null) {
+            return null;
+        }
+        
+        return findRootNodeInMid(head,null);
+    }
+    
+    public TreeNode findRootNodeInMid(ListNode head, ListNode tail){
+        ListNode slow = head;
+        ListNode fast = head;
+        
+        if(head==tail) {
+            return null;
+        }
+    
+        while(fast!=tail&&fast.next!=tail){
```

Comment:
hmm, i think appropriate style helps others, including interviewers, to understand in a short amount of time (avoid visual fatigue). maybe we can try to apply google java style guide in future prs?
https://google.github.io/styleguide/javaguide.html